src/core/eval.ml

```diff
@@ -192,9 +192,11 @@ and tree_walk : dtree -> ctxt -> stack -> (term * stack) option =
     let open Tree_types in
     match tree with
     | Fail                                                -> None
-    | Leaf(env_builder, act)                              ->
-        (* Allocate an environment for the action. *)
-        let env = Array.make (Bindlib.mbinder_arity act) TE_None in
+    | Leaf(env_builder, act, xvars)                       ->
```

Comment:
@gabrielhdt Could you please remind me and update tree_types.ml of what is the meaning of the elements of the first argument of Leaf?